webapp/channels/src/components/admin_console/data_grid/data_grid_header.tsx

```diff
@@ -12,13 +12,12 @@ export type Props = {
     columns: Column[];
 }
 
-class DataGridHeader extends React.Component<Props> {
-    renderHeaderElement(col: Column) {
+const DataGridHeader = ({columns}: Props) => {
+    const renderHeaderElement = (col: Column) => {
```

Comment:
The `renderHeaderElement` is a component itself, and doesn't need any of the props, so we could move that out of the component. So... we would have the code something like this:

```
const HeaderElement = ({col}: HeaderElementProps) => {
    ...
}
const DataGridHeader = ({columns}: Props) => {
    return (
        <div className='DataGrid_header'>
            {columns.map((col) => <HeaderElement col=col>)}
        </div>
    )
}
```

Why this is better? If you declare the function outside of the component, it gets instantiated just once during the lifetime of the program. Otherwise, on every render of every data grid header you will instantiate the function and potentially remove all references at some point, which consumes extra memory and generates extra work for the GC.

Sorry for the back and forth with this. I just saw this approach on a different PR and realized it is something nice to do here too :smile: